cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/DefaultPreviewManager.java

```diff
@@ -115,13 +119,20 @@
     this.secureStore = secureStore;
     this.transactionSystemClient = transactionSystemClient;
     this.previewDataDir = Paths.get(cConf.get(Constants.CFG_LOCAL_DATA_DIR), "preview").toAbsolutePath();
-    this.maxPreviews = cConf.getInt(Constants.Preview.CACHE_SIZE, 10);
+    this.maxConcurrentPreviews = cConf.getInt(Constants.Preview.CACHE_SIZE, 10);
     this.previewRunnerModuleFactory = previewRunnerModuleFactory;
+    this.previewPollers = new HashSet<>();
   }
 
   @Override
   protected void startUp() throws Exception {
     previewInjector = createPreviewInjector();
+    // Create and start the preview poller services.
+    for (int i = 0; i < maxConcurrentPreviews; i++) {
+      Service pollerService = previewInjector.getInstance(PreviewRunnerService.class);
```

Comment:
The PreviewManager shouldn't be starting this service. Is this only used in standalone?

Reply:
Yes. We will have another preview manager implementation for distributed which wont start these services.

Reply:
Moved to separate service `PreviewRunnerManager`